src/c/types/raylib_types.ml

```diff
@@ -1,593 +1,292 @@
-
-    let () = seal t
-  end
-
-  module Model = struct
-    type t = [ `Model ]
-
-    let t : t structure typ = structure "Model"
-
-    let transform = field t "transform" Matrix.t
-
-    let mesh_count = field t "meshCount" int
-
-    let meshes = field t "meshes" (ptr Mesh.t)
-
-    let material_count = field t "materialCount" int
-
-    let materials = field t "materials" (ptr Material.t)
-
-    let bone_count = field t "boneCount" int
-
-    let bones = field t "bones" (ptr BoneInfo.t)
-
-    let bind_pose = field t "bindPose" (ptr Transform.t)
-
-    let () = seal t
-  end
-
-  module ModelAnimation = struct
-    type t = [ `ModelAnimation ]
-
-    let t : t structure typ = structure "ModelAnimation"
-
-    let bone_count = field t "boneCount" int
-
-    let bones = field t "bones" (ptr BoneInfo.t)
-
-    let frame_count = field t "frameCount" int
-
-    let frame_poses = field t "framePoses" (ptr (ptr Transform.t))
-
-    let () = seal t
-  end
-
-  module Ray = struct
-    type t = [ `Ray ]
-
-    let t : t structure typ = structure "Ray"
-
-    let position = field t "position" Vector3.t
-
-    let direction = field t "direction" Vector3.t
-
-    let () = seal t
-  end
-
-  module RayHitInfo = struct
-    type t = [ `RayHitInfo ]
+module Vector2 = struct
+  type%c t = { x : float; y : float } [@@cname "Vector2"]
+end
 
-    let t : t structure typ = structure "RayHitInfo"
+module Vector3 = struct
+  type%c t = { x : float; y : float; z : float } [@@cname "Vector3"]
+end
 
-    let hit = field t "hit" bool
+module Vector4 = struct
+  type%c t = { x : float; y : float; z : float; w : float } [@@cname "Vector4"]
+end
 
-    let distance = field t "distance" float
+module Quaternion = struct
+  type t = Vector4.t
 
-    let position = field t "position" Vector3.t
+  let%c t = Vector4.t
+end
 
-    let normal = field t "normal" Vector3.t
+module Matrix = struct
+  type%c t = {
+    m0 : float;
+    m1 : float;
+    m2 : float;
+    m3 : float;
+    m4 : float;
+    m5 : float;
+    m6 : float;
+    m7 : float;
+    m8 : float;
+    m9 : float;
+    m10 : float;
+    m11 : float;
+    m12 : float;
+    m13 : float;
+    m14 : float;
+    m15 : float;
+  }
+  [@@cname "Matrix"]
+end
 
-    let () = seal t
-  end
+module Color = struct
+  type%c t = { r : uchar; g : uchar; b : uchar; a : uchar } [@@cname "Color"]
+end
 
-  module BoundingBox = struct
-    type t = [ `BoundingBox ]
+module Rectangle = struct
+  type%c t = { x : float; y : float; width : float; height : float }
+  [@@cname "Rectangle"]
+end
 
-    let t : t structure typ = structure "BoundingBox"
+module Image = struct
+  type%c t = {
+    data : void ptr;
+    width : int;
+    height : int;
+    mipmaps : int;
+    format : int;
+  }
+  [@@cname "Image"]
+end
 
-    let min = field t "min" Vector3.t
+module Texture2D = struct
+  type%c t = {
+    id : uint;
+    width : int;
+    height : int;
+    mipmaps : int;
+    format : int;
```

Comment:
This was the main reason, I didn't go with pxx_cstubs the first time, the format being `int` and not `PixelFormat.t`. But since these types are wrapped anyway in raylib.ml, I guess it doesn't really matter. For the user the format will always be `PixelFormat.t`

Reply:
Yeah it's a bit annoying. We could put the constants in the same file and make a Ctypes `view` over an int, but that's a lot of work for something that doesn't change the user's interaction with the library.